dynamic_clamp/dynamic_clamp.ino

```diff
@@ -72,102 +110,92 @@ elapsedMillis epscTime;           // msec, time since last EPSC was triggered
-  pinMode(epscTriggerPin,INPUT);                // define the pin that receives a trigger when an EPSC should be injected
+  pinMode(PIN_EPSC_TRIGGER, INPUT);             // define the pin that receives a trigger when an EPSC should be injected
 }
 
-
-
+// vars to decode params sent from GUI
+float dataTemp[nPars] = {0.00};
+int parNo = 0;
+float tempVal;
+char tempBytes[4];
 
 void loop() {
-
-  // Part 1:    Serial Communication & Parameter Setting
-  // The host computer tells the Teensy when the dynamic clamp parameters should change.
-  // To do this, it sends 4-byte-long numbers, one for each parameter. So if there are 
-  // 8 adjustable parameters, it will send 32 bytes. At present, the adjustable parameters
-  // are the shunt conductance (nS), maximal HCN conductance (nS), maximal Na conductance (nS),
-  // excitatory OU conductance (OU1) mean (nS), excitatory OU conductance diffusion constant (nS^2/ms),
-  // inhibitory OU conductance (OU2) mean (nS), inhibitory OU conductance (OU2) diffusion
-  // constant (nS^2/ms), and maximal EPSC conductance (nS).
-  if (p0 > pollInterval) {                                            // check for conductance updates frequently
-    union {
-      byte asByte[4];
-      float asFloat;
-    } data1;
-    static float dataTemp[nPars] = {0.0};
-    static byte parNo = 0;
-    if (Serial.available()>3) {
-      for (int x=0; x<4; x++) data1.asByte[x] = Serial.read();
-      dataTemp[parNo] = data1.asFloat;
-      parNo++;
-      if (parNo==nPars) {
-        gShunt = dataTemp[0];
-        gH = dataTemp[1];
-        gNa = dataTemp[2];
-        OU1_mean = dataTemp[3];
-        OU1_D = dataTemp[4];
-        OU2_mean = dataTemp[5];
-        OU2_D = dataTemp[6];
-        gEPSC = dataTemp[7];
-        parNo = 0;
-        for (int n=0; n<nPars; n++) Serial.println(dataTemp[n]);      // (prepare) data echo to sender
-        Serial.send_now();                                            // schedule immediate transmission
-      }
+    // Part 1:    Serial Communication & Parameter Setting
+    // The host computer tells the Teensy when the dynamic clamp parameters should change.
+    // To do this, it sends 4-byte-long numbers, one for each parameter. So if there are
+    // 8 adjustable parameters, it will send 32 bytes. At present, the adjustable parameters
+    // are the shunt conductance (nS), maximal HCN conductance (nS), maximal Na conductance (nS),
+    // excitatory OU conductance (OU1) mean (nS), excitatory OU conductance diffusion constant (nS^2/ms),
+    // inhibitory OU conductance (OU2) mean (nS), inhibitory OU conductance (OU2) diffusion
+    // constant (nS^2/ms), and maximal EPSC conductance (nS).
+    if (p0 > pollInterval) {
+
+        while (Serial.available()) {
+            Serial.readBytes(tempBytes, 4);
+            memcpy(&tempVal, &tempBytes, 4);
+            dataTemp[parNo] = tempVal;
+            parNo++;
+        }

```

Comment:
used some built-in features to make reading the encoded floats cleaner with some acceptable assumptions.
- we can assume that if data is available that it is at least one float value since `processing_control` does not send complex data of different lengths - so we can read 4 bytes at once
- using `union` type conversion is [not safe](https://learn.adafruit.com/introducing-adafruit-itsybitsy-m4/adapting-sketches-to-m0#aligned-memory-access-2677101) for 32 bit processors as memory architecture may not allow it. `memcpy` is always safe.